Add release notes for 6.10 (#663)
chef/cookstyle: RELEASE_NOTES.md

## Cookstyle 6.10

### 3 New Cops

#### ChefSharing/IncludeResourceExamples

> The `ChefSharing/IncludeResourceExamples` cop detects resources that don't include examples. The `chef-resource-inspector` command displays the examples in resources and that data can be used to automatically generate documentation.

Reviewer: ```suggestion
The `ChefSharing/IncludeResourceExamples` cop detects resources that don't include examples. The `chef-resource-inspector` command displays resource examples and those examples can be used to automatically generate documentation.
```
